### ISSUE TITLE
Could not connect to iscsid - Initiator configuration (comment)

### ISSUE THREAD

**Issue by san-nas-san** [iscsi-osx/iSCSIInitiator#54]
Hi nsinenian,

I couldn't connect to icsid - when I tried to setup  according to your wiki **Initiator configuration**.

After a lot of searching I found out that you have to use terminal with **sudo** for the commands necessary for setting up.

Could you please update or add this in  your wiki accordingly, as it would have saved me a lot of time.
I'm using the initiator on OSX Mavericks and a ReadyNAS OS 6

:-)

---

in the WIKI

> The initiator's iSCSI Qualified Name (IQN) and alias should be configured after installation. Default values for the IQN and alias are set by the initiator which may not be unique in the iSCSI domain. The IQN can be modified as follows:
> 
> $ iscsictl modify initiator-config -node-name [IQN]
> 
> Where [IQN] is specified as defined in RFC3720 (see iSCSI overview). The initiator alias may similarly be configured:
> 
> $ iscsictl modify initiator-config -node-alias [alias]
> 
> These values are part of the initiator configuration that can be viewed using the list subcommand:

It should be: 
">$ **sudo** iscsictl modify initiator-config -node-name [IQN]"

**Comment by nsinenian** [comment]
"sudo" is no longer required for beta-3. Are you using an older version (beta2)? The current documentation reflects the latest release so it is correct.

**Comment by san-nas-san** [comment]
Thanks, yes you were partially right - I still had the beta 2, which I downloaded shortly before you published the beta3. 
I deinstalled with the beta2 deinstaller and therafter installed the beta3 - but still having  issues. I still get 
`iscsictl: Could not connect to iscsid.` if I don't use  sudo.

With sudo I can sometimes mount the target

Moreover the ISCSI Target disconnects sometimes and causes my MacbookPro to freeze with the screen turning lighgrey - like in the startup screen - but no wheel or anything - needs to be hard resetted then.
I'm  running your beta 3 under OS X  10.9.5 connecting to an iSCSI share on a Netgear on the ReadyNAS

I will give it a try again in some days

Thanks for your effort and developing an iSCSI initiator for  OSX

**Comment by nsinenian** [comment]
I'm unable to reproduce your problem - and it sounds like there might be two problems. Can you please provide as much of following:

1. OS X version number
2. Permissions on the iscsid socket (run "ls -l /var/run/iscsid" at the terminal and send me the output).
3. Any kernel crash logs - when the target disconnects and turns the screen light gray, upon reboot, please go to the the "Console" application and look for any kernel crash logs under "System Diagnostic Reports" - the date/time stamp should match.

Thanks!

**Comment by san-nas-san** [comment]
Hello nsinenian, thanks for your reply
Information that might be helpful:
- OSX version number is 10.9.5, MacbookPro, the iSCSI LUN is located on a ReadyNAS 316, without CHAP, any initiator accepted, connection is via WLAN/WiFi
- Permissions on the iscsid socket:
yogis-mbp-25:~ root# ls -l /var/run/iscsid
srwx------  1 root  daemon  0 Jul  2 23:46 /var/run/iscsid

- kernel crash log as follows - the only one I could find

[Uploading Kernel_2016-06-26-214939_Yogis-MacBook-Pro.panic.zip…]()



Thank you for your immediate response, time and work that you put into the iSCSIInitiator

_P.S.
-- in the meantime I was able to connect, but the iSCSI Drive disconnects unexpectedly after at most an hour although the MacbookPro hasn't moved and been turned on all the time & didn't go to sleep
-- I will monitor the tcp stream the next time when I try your initiator again  - unfortunately I'm very busy next week and won't have the time to experiment with the iniatiator.
-- I will pay attention the next time a iSCSI associated crash occurs and look at the logs_

**Comment by nsinenian** [comment]
@san-nas-san Any update on this issue? Does beta-4 work better? Please let us know!

**Comment by san-nas-san** [comment]
Thanks for the reminder.
Still having issues that the wifi-connection within the home-network  is lost after an unspecified time — all energy saving options / sleep-settings etc. are turned off. Time that suddenly the mounted iscsi suddenly disappears and e.g. copy procedures are interrupted harshly can occur after 30 Minutes up to something like 17 hours. 

I will download the beta-4 and give it a try


Am 08.09.2016 um 12:48 schrieb Nareg Sinenian <notifications@github.com>:

**Comment by appoli13** [comment]
I only recently installed iscsi initiator, but I had the issue of a dropped connection when I didn't actually close my computer (put it to sleep). I checked the power management config (pmset) & noticed that my computer was set to sleep when plugged into AC. I turned off sleep when plugged into AC & left my computer on overnight connected to an iscsi drive - in the morning the drive was still mounted.

While in terminal I checked the pmset log and noticed that it looked like when the computer was inactive and it was trying to go to sleep (I believe it was checking if any connections were still active) I noticed that it had said iscsi had already timed out (>3000 ms) & was only waiting for some other activities that were slowing down. Do you think there is anything that could be done to address this? Would setting the refresh interval down to 1 help?

Ideally I would like to have the initiator maintain a connection in a low power/sleep state so that I can re-enable sleep.

The following should most likely be on another thread, but it's somewhat related:
I ran the script to instruct iscsi initiator to auto-connect to a target (and get confirm that it ran OK), but when I power up my machine it doesn't connect (not the biggest deal, just created a couple automator services for login/logout). I mention this b/c maybe a quick workaround for the above would be to logout when sleep was about to happen & perform a login when the computer wakes?

I also noticed that the initial config changes I made don't seem to stick when I reboot. The targets i've specified and the refresh instructions stick, but the name & alias don't (maybe this is why it doesn't auto connect?)

**Comment by nsinenian** [comment]
First, are you using Sierra or El Capitan? Sierra has not been tested and may have issues.

> I only recently installed iscsi initiator, but I had the issue of a dropped connection when I didn't actually close my computer (put it to sleep). I checked the power management config (pmset) & noticed that my computer was set to sleep when plugged into AC. I turned off sleep when plugged into AC & left my computer on overnight connected to an iscsi drive - in the morning the drive was still mounted.

So what you're saying is that the connection dropped when your machine went to sleep, and it never came back?

> While in terminal I checked the pmset log and noticed that it looked like when the computer was inactive and it was trying to go to sleep (I believe it was checking if any connections were still active) I noticed that it had said iscsi had already timed out (>3000 ms) & was only waiting for some other activities that were slowing down. Do you think there is anything that could be done to address this? Would setting the refresh interval down to 1 help?

I'm not sure that I understand the problem - are you saying that the system was waiting on iSCSI to sleep and it took more than 3 seconds? Can you provide the actual logs? As far as I know, iSCSI hasn't preventing a machine from going to sleep (no one has reported this) and I am unable to reproduce this. If iSCSI has trouble logging out of targets when the machine wants to sleep (or if logout takes time) then this may occur. But that's really a function of your iSCSI setup (e.g., slow target to respond or connection or multiple targets that need to be logged out). In addition, it takes time for the OS to unmount LUNs (think about how there's a 1/2 to 1 sec delay when you eject a USB drive for example).

>  Ideally I would like to have the initiator maintain a connection in a low power/sleep state so that I can re-enable sleep.

As far as I know, this is not possible. The current mac OS API exposes certain power management functions that iSCSI currently uses. Keeping things alive while in lower power or sleep mode is not an option that a developer can exploit (except for Apple, which is why it can check for mail and the like while in "power nap" mode). For instance, the OS can decide to shut down an interface during low power mode and there's nothing we can do about it. If this option becomes available to us in the future we'll implement it.

> The following should most likely be on another thread, but it's somewhat related:
I ran the script to instruct iscsi initiator to auto-connect to a target (and get confirm that it ran OK), but when I power up my machine it doesn't connect (not the biggest deal, just created a couple automator services for login/logout). I mention this b/c maybe a quick workaround for the above would be to logout when sleep was about to happen & perform a login when the computer wakes?

This is exactly how iSCSI works now. It actually gets notification from the kernel when the machine wants to sleep. It then records all of the connections, logs out of all of them, and queues them for login when the network interface becomes available again (upon restore from sleep). Several users have reported that this works, and I have not been able to produce a failure on my machine yet. 

A few questions / suggestions to get to the bottom of this:

Are you using a wired or wireless connection?
Can you connect to a target and force the machine to sleep (Apple Menu -> Sleep) and confirm that you get the same result as when the machine sits inactive?
Look at the system console and let me know what it says before/after sleep. Under the "FILES" section, the file "system.log" contains messages printed by iscsid. Let me know what you see there.

> I also noticed that the initial config changes I made don't seem to stick when I reboot. The targets i've specified and the refresh instructions stick, but the name & alias don't (maybe this is why it doesn't auto connect?)

There is nothing special about initiator name and alias - it is handled like other options. After changing these options, please look at the contents of `/Library/Preferences/com.github.iscsi-osx.iSCSIInitiator.plist` and verify that the name/alias in fact changed. Then, reboot and check the file again to ensure that changes did stick. Either way, let me know - that will help identify where the problem lies.

**Comment by appoli13** [comment]
Hi,

Apologies for any confusion!

1. I was using El Capitan at the time, just upgraded to Sierra yesterday & all the testing I had done was on El Capitan
2. iscsi did NOT prevent the computer from going to sleep. Just the opposite! I was hoping it COULD prevent the computer from going to sleep! (that way when I have the drive mounted I don't need to worry about turning sleep off)
Thus the post regarding refresh interval - I was hoping that maybe if I increased the frequency the kernel wouldn't detect that iscsi was timing out (for machine sleep purposes)
3. When I boot my machine the iscsi doesn't automatically mount the image. When the machine goes to sleep I usually get a notification saying the drive was disconnected without ejecting it before hand & iscsi does NOT re mount the drive

I will check the plist file when I get home later today & would be happy to upload any logs (although I'm not sure if they are necessary since iscsi did not prevent the machine from going to sleep - i turned sleep off since iscsi was not ejecting properly/reconnecting when the machine woke up).

thank you for your help!

oh and your questions:
1. since this is new software to me I'm sticking with wired connection only (help troubleshoot any initial glitches)
2. I just forced the machine to sleep - it didn't give me the warning that the drive had been improperly disconnected, but the drive did not reconnect on its own.

as i mentioned - logs will be posted later today
